libraries/apollo-normalized-cache-api/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/api/OperationCacheExtensions.kt

```diff
@@ -69,7 +69,7 @@ private fun <D : Executable.Data> Executable<D>.readInternal(
       cache = cache,
       cacheHeaders = cacheHeaders,
       cacheResolver = cacheResolver,
-      variables = variables(customScalarAdapters),
+      variables = variables(customScalarAdapters, true),
```

Comment:
Also means we're introducing a cache miss for anyone who has relied on `null` being in the cache key for boolean values with a a default (because now it will be the default value instead of `null`). Soooooo we might want to revert the revert and keep `default` value in CompiledField, at least for 3.x